drivers/spi/spi_ll_stm32.c

```diff
@@ -57,17 +57,16 @@ static int spi_stm32_get_err(SPI_TypeDef *spi)
 
 static inline u16_t spi_stm32_next_tx(struct spi_stm32_data *data)
 {
-	u16_t tx_frame;
+	u16_t tx_frame = SPI_STM32_TX_NOP;
 
-	if (spi_context_tx_on(&data->ctx)) {
+	if (data->ctx.tx_buf) {
```

Comment:
Check PR #5785 buf can be non-null, but if len is 0 we should not touch it.

So instead, use spi_context_tx_buf_on()